### PR TITLE
fix: return default logger

### PR DESCRIPTION
**Pull request by vikkymanit**
Return a default logger and fixing panic when trying to get a logger without initialisation.

### CODE REVIEW

**Comment by AttilaW-Keytiles** [comment]
The changes now look good!
but...

It deviates from the main branch - e.g. CHANGELOG is completely invalid now

I would recommend to fork a new branch out from main now, apply changes there, and open a new PR
we could take that
And then we could close these PRs

File: ktlogging.go [hunk 1]
```diff
@@ -79,6 +79,14 @@ func With(loggerName string) *Logger {
 
 // internal method to get a logger - NOT THREAD SAFE! Already assumes Lock is established so no race condition!
 func getLogger(loggerName string) *Logger {
+	if loggers == nil {
```
Comment (AttilaW-Keytiles) [inline ktlogging.go:82]: looks good

File: CHANGELOG.md [hunk 2]
```diff
@@ -10,6 +10,12 @@ This code is representing version `1.0.0`
 
 # Changes in releses
 
+## release 1.0.1
```
Comment (AttilaW-Keytiles) [inline CHANGELOG.md]: this is deviating a lot from current main branch so see my in general comment!

File: ktlogging.go [hunk 3]
```diff
@@ -106,13 +114,31 @@ func getLogger(loggerName string) *Logger {
...
-		// OK let's create a silent "root" logger programmatically (this will not log anywhere)
-		rootLogger = &Logger{name: "root", level: InfoLevel, handlers: map[string]*zap.Logger{}}
-		loggers[_ROOT_NAME] = rootLogger
+		// this should never happen, as a root logger should have been created if loggers were not initialised
+		panic("root logger not found! loggers initialisation likely did not happen correctly.")
 	}
 	return rootLogger
 }
 
+func getDefaultLoggerConfig() ConfigModel {
```
Comment (AttilaW-Keytiles) [inline ktlogging.go:123]: also looks good

**Review by AttilaW-Keytiles — CHANGES_REQUESTED** [review]
all good - but see my main comment in conversation

**Comment by vikkymanit** [comment]
I have merged the latest from main onto this branch. So we should be fine now.

File: CHANGELOG.md (hunk 2 above)
Comment (vikkymanit) [reply CHANGELOG.md]: I have merged the latest from main onto this branch. So we should be fine now.

**Review by AttilaW-Keytiles — APPROVED** [review]
looks good - lets merge and move on

**Pull request merged**